### PR TITLE
[INFRA-7354] Check schedule iCal URL in update_schedule_problems

### PR DESCRIPTION
**Pull request by Aloshi**
This check distinguishes between three possible problems with the iCal URL:
* The iCal URL is empty (treated as a problem)
* The iCal URL is broken (connection error, HTTP 4xx/5xx, ...)
  - `Schedule.get_calendar_data()` now checks for this using `requests.Response.raise_for_status()`
* The iCal URL isn't returning valid iCal data (failing to parse)
  - For example, if someone entered `http://google.com`or a link to another calendar format

There are also unit tests for each case.

---

Also, this commit splits up the "update shifts" and "update schedule problems" logic into two independent try/except blocks. Previously, if updating shifts failed, update problems would never run (!!).

### CODE REVIEW

File: cabot/cabotapp/schedule_validation.py [hunk 1]
```diff
@@ -55,18 +56,33 @@ def _find_problems(schedule, current_time=None):
...
         problems.append("The schedule has no fallback officer.")
     elif not schedule.fallback_officer.email:
         problems.append("The fallback officer does not have an email set.")
 
+    # test the iCal url
+    if schedule.ical_url:
+        try:
+            schedule.get_calendar_data()
+        except requests.RequestException as e:
+            problems.append("The schedule's iCal URL returns an error ({}).".format(e))
```
Comment (elainearbaugh) [inline cabot/cabotapp/schedule_validation.py]: nit: maybe "The schedule's iCal URL is invalid"? "returns an error" is kinda confusing imo

**Comment by Aloshi** [comment]
(dismissed due to rebase, going to force merge after tests pass)

**Comment by Aloshi** [comment]
trigger build

**Comment by Aloshi** [comment]
trigger build

**Pull request merged**